rbac/internal/views.py

```diff
@@ -781,3 +781,53 @@ def roles(request, uuid: str) -> HttpResponse:
 def trigger_error(request):
     """Trigger an error to confirm Sentry is working."""
     raise SentryDiagnosticError
+
+
+def correct_resource_definitions(request):
+    """Get/Fix resourceDefinitions with incorrect attributeFilters.
+
+    Attribute filters with lists must use 'in' operation. Those with a single string must use 'equal'
+
+    GET /_private/api/utils/resource_definitions
+    PATCH /_private/api/utils/resource_definitions
+    """
+    list_query = """ FROM management_resourcedefinition
+                WHERE "attributeFilter"->>'operation' = 'equal'
+                AND jsonb_typeof("attributeFilter"->'value') = 'array';"""
+
+    string_query = """ from management_resourcedefinition WHERE "attributeFilter"->>'operation' = 'in'
+                AND jsonb_typeof("attributeFilter"->'value') = 'string';"""
+
+    if request.method == "GET":
+        with connection.cursor() as cursor:
+
+            cursor.execute("SELECT COUNT(*)" + list_query)
+            count = cursor.fetchone()[0]
+
+            cursor.execute("SELECT COUNT(*)" + string_query)
+            count += cursor.fetchone()[0]
+
+        return HttpResponse(f"{count} resource definitions would be corrected", status=200)
+    elif request.method == "PATCH":
+        count = 0
+        with connection.cursor() as cursor:
+
+            cursor.execute("SELECT id " + list_query)
+            result = cursor.fetchall()
+            for id in result:
+                resource_definition = ResourceDefinition.objects.get(id=id[0])
+                resource_definition.attributeFilter["operation"] = "in"
+                resource_definition.save()
+                count += 1
+
+            cursor.execute("SELECT id " + string_query)
+            result = cursor.fetchall()
+            for id in result:
+                resource_definition = ResourceDefinition.objects.get(id=id[0])
+                resource_definition.attributeFilter["operation"] = "equal"
+                resource_definition.save()
```

Comment:
There shouldn't be a ton of these cases, so performance should be acceptable

Reply:
I did check our DB's for these cases and there were only 2 or 3 that would need a change.